### DISCUSSION TITLE
Apologies...not sure how to make a ticket.

### DISCUSSION THREAD

**Discussion by RayMan412** [GerryDazoo/Slinger#66] [General]
Gerry, who is awesome, said for me to make a ticket so he could help me.
I assume this is what it is...

I think I am getting close.

When I run the program I get this:

C:\Slingbox Server>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.04 Running on Windows-10-10.0.19043-SP0
Using config file config.ini
Connection Manager Running on port 5001 with 10 max streams....
Streamer Running:  10 config.ini 5001
starting up on  port 5001
Checking for slingbox at ('http://192.168.1.15', 5001)
Error connecting to  ('http://192.168.1.15', 5001)
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:5002
 * Running on http://192.168.1.52:5002 (Press CTRL+C to quit)
No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::8df4:f798:4020:2a1b%24
Finding Slingbox on local network. My IP Info =  10.21.242.99
09/13/2022, 16:17:31  Registering Slingboxes {'395808DB6C4591CE3DFA8E85F3309883': 5001}
Finding Slingbox on local network. My IP Info =  fe80::65ca:5bde:4401:4ff2%19
Finding Slingbox on local network. My IP Info =  fe80::f1:60fc:c908:8402%8
Finding Slingbox on local network. My IP Info =  fe80::15a8:c1e:f1cf:50c5%16
Finding Slingbox on local network. My IP Info =  fe80::d46e:5eb8:349c:78b7%20
Finding Slingbox on local network. My IP Info =  192.168.1.52
Slingbox Found 192.168.1.15 5201 " New York HD Slingbox             "
Finding Slingbox on local network. My IP Info =  fe80::5d3c:7f28:4d0d:1303%6
Finding Slingbox on local network. My IP Info =  ::1
Finding Slingbox on local network. My IP Info =  127.0.0.1
Using slingbox at  ('192.168.1.15', 5201)
Streamer:  New York HD Slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

My Config File:
name=New York HD Slingbox

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=30

;Min 50 Max 8000
VideoBandwidth=2000

; Min 0 (auto) max 63
VideoSmoothness=63

; Send Iframe every n seconds Max 30
IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, ???
;Solo, M1 0=composite 1=component
;500  0=composite, 1=component, 2=HDMI
; If you don't specify it will default to the last selected input
; So if you only have one video source connected you don't need to 
; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter example
VideoSource=1

[SERVER]
; local port number for the server to listen on for connections
port=5001
maxstreams=10
enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work
;code=2
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
include=remote.txt 

Any help is appreciated. 

Ray "NFLFAN412"

**Comment by GerryDazoo** [comment]
The code managed to find your slingbox on your network  ('192.168.1.15', 5201) even though it was mis-configured in your config.ini. So it looks like it's waiting for you to make an http streaming connection from your favourite media player. So I'm not sure what you think needs a ticket.

A few notes:
1) Get the latest code R3.07.
2) You configured the wrong IP address and port number for your slingbox in your config.ini 192.168.1.15, 5001 the correct info is  ('192.168.1.15', 5201)
2) Add the slingbox ip address and port number in your config.ini. This will avoid having the server scan for your slingbox on every startup.
3) Don't use a slingbox-ish port number, 5001, for the server. It just causes confusion. Unless you've got a good reason to change it the default 8080 should work fine.
4) Did you miss some lines from your config file? [SLINGBOX] . You're missing the slingbox type (defaults to 500) and the slingbox password and the slingbox IP info, did you leave that out on purpose?
5)the code in the [REMOTE] section should be the same as your SB video source. Check the latest config.ini for some rudimentary documentation. In your case based on your supplied config file it should be "1".

**Comment by RayMan412** [comment]
Ok..
Hopefully getting Closer. Full Config file (I had it cut off last time)
[SLINGBOX]sbtype=350;sbtype="Solo/Pro/ProHD"password=slingbox; uncomment the next lines ipaddress and port with your local network info; if you don't want the server to automatically find your ; slingbox on the local network. Needed if server and slingbox are; not on the same LAN segment. ALso removes requirement for the netifaces module  ipaddress=http://192.168.1.15port=5201; Repace with your finderidfinderid=395808DB6C4591CE3DFA8E85F3309883
name=New York HD Slingbox
; Following are the default values; Default resolution when server starts. You can change this and reconnect without having to restart the server; valid range = 1..15Resolution=12
; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolutionFrameRate=30
;Min 50 Max 8000VideoBandwidth=2000
; Min 0 (auto) max 63VideoSmoothness=63
; Send Iframe every n seconds Max 30IframeRate=5
; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32AudioBitRate=32
; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of;Composite, Component, HDMI or Tuner.;ProHD 0=Tuner, ???;Solo, M1 0=composite 1=component;500  0=composite, 1=component, 2=HDMI; If you don't specify it will default to the last selected input; So if you only have one video source connected you don't need to ; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter exampleVideoSource=1
[SERVER]; local port number for the server to listen on for connectionsport=8080maxstreams=10enableremote=yes
[REMOTE]; see release notes for changing this if your remote doesn't workcode=1; path to an external remote control definitions file, if the defaults; aren't working for you or you want to change the format of the web page.; use the supplied remote.txt as a starting point.include=remote.txt 

Server Says:
Version : 3.07 Running on Windows-10-10.0.19043-SP0 pid= 26332Using config file config.iniConnection Manager Running on port 8080 with 10 max streams....Streamer Running:  10 config.ini 8080starting up on  port 8080Checking for slingbox at ('http://192.168.1.15', 5201)Error connecting to  ('http://192.168.1.15', 5201)Reading Custom Remote definition from remote.txt * Serving Flask app 'slingbox_server' (lazy loading) * Environment: production   WARNING: This is a development server. Do not use it in a production deployment.   Use a production WSGI server instead. * Debug mode: off * Running on all addresses (0.0.0.0)   WARNING: This is a development server. Do not use it in a production deployment. * Running on http://127.0.0.1:8081 * Running on http://192.168.1.21:8081 (Press CTRL+C to quit)No valid slingbox ip info found in config.iniFinding Slingbox on local network. My IP Info =  fe80::8df4:f798:4020:2a1b%25Finding Slingbox on local network. My IP Info =  10.21.242.9909/15/2022, 15:58:52  Registering Slingboxes {'395808DB6C4591CE3DFA8E85F3309883': 8080}Getting external IP addressRedirector, my external IP address 67.82.153.250Finding Slingbox on local network. My IP Info =  fe80::65ca:5bde:4401:4ff2%20Finding Slingbox on local network. My IP Info =  fe80::f1:60fc:c908:8402%8Finding Slingbox on local network. My IP Info =  fe80::15a8:c1e:f1cf:50c5%17Finding Slingbox on local network. My IP Info =  fe80::d46e:5eb8:349c:78b7%21Finding Slingbox on local network. My IP Info =  192.168.1.21Slingbox Found 192.168.1.15 5201 " New York HD Slingbox             "Finding Slingbox on local network. My IP Info =  fe80::5d3c:7f28:4d0d:1303%6Finding Slingbox on local network. My IP Info =  ::1Finding Slingbox on local network. My IP Info =  127.0.0.1PING failure ('67.82.153.250', 8080) [WinError 10061] No connection could be made because the target machine actively refused itNo working external portmaps. Not registering slingboxesRetrying in 10 minutesUsing slingbox at  ('192.168.1.15', 5201)Streamer:  New York HD Slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox


-----Original Message-----

**Reply by GerryDazoo** [reply]
That all looks good. You can comment out your finderid for now.
The server is waiting for you to make a connection request from your favourite media player.

**Comment by RayMan412** [comment]
So I try to connect  and I get 





Here's the VLC error log-- logger module started --main: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.main: playlist is emptyaccess error: HTTP connection failuremain error: read error: No errorhttp error: failed to read answeraccess error: HTTP connection failurehttp error: cannot connect to 192.168.1.15:5201-- logger module stopped --


Here is my Config again:[SLINGBOX]sbtype=350;sbtype="Solo/Pro/ProHD"password=slingbox; uncomment the next lines ipaddress and port with your local network info; if you don't want the server to automatically find your ; slingbox on the local network. Needed if server and slingbox are; not on the same LAN segment. ALso removes requirement for the netifaces module  ipaddress=http://192.168.1.15port=5201; Repace with your finderidfinderid=395808DB6C4591CE3DFA8E85F3309883
name=New York HD Slingbox
; Following are the default values; Default resolution when server starts. You can change this and reconnect without having to restart the server; valid range = 1..15Resolution=12
; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolutionFrameRate=30
;Min 50 Max 8000VideoBandwidth=2000
; Min 0 (auto) max 63VideoSmoothness=63
; Send Iframe every n seconds Max 30IframeRate=5
; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32AudioBitRate=32
; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of;Composite, Component, HDMI or Tuner.;ProHD 0=Tuner, ???;Solo, M1 0=composite 1=component;500  0=composite, 1=component, 2=HDMI; If you don't specify it will default to the last selected input; So if you only have one video source connected you don't need to ; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter exampleVideoSource=1
[SERVER]; local port number for the server to listen on for connectionsport=8080maxstreams=10enableremote=yes
[REMOTE]; see release notes for changing this if your remote doesn't workcode=1; path to an external remote control definitions file, if the defaults; aren't working for you or you want to change the format of the web page.; use the supplied remote.txt as a starting point.include=remote.txt 




Here is my Slinger:Version : 3.07 Running on Windows-10-10.0.19043-SP0 pid= 28928Using config file config.iniConnection Manager Running on port 8080 with 10 max streams....Streamer Running:  10 config.ini 8080starting up on  port 8080Checking for slingbox at ('http://192.168.1.15', 5201)Error connecting to  ('http://192.168.1.15', 5201)Reading Custom Remote definition from remote.txt * Serving Flask app 'slingbox_server' (lazy loading) * Environment: production   WARNING: This is a development server. Do not use it in a production deployment.   Use a production WSGI server instead. * Debug mode: off * Running on all addresses (0.0.0.0)   WARNING: This is a development server. Do not use it in a production deployment. * Running on http://127.0.0.1:8081 * Running on http://192.168.1.21:8081 (Press CTRL+C to quit)No valid slingbox ip info found in config.iniFinding Slingbox on local network. My IP Info =  fe80::65ca:5bde:4401:4ff2%20Finding Slingbox on local network. My IP Info =  fe80::f1:60fc:c908:8402%8Finding Slingbox on local network. My IP Info =  fe80::15a8:c1e:f1cf:50c5%17Finding Slingbox on local network. My IP Info =  fe80::d46e:5eb8:349c:78b7%21Finding Slingbox on local network. My IP Info =  192.168.1.2109/15/2022, 22:12:51  Registering Slingboxes {'395808DB6C4591CE3DFA8E85F3309883': 8080}Getting external IP addressSlingbox Found 192.168.1.15 5201 " New York HD Slingbox             "Redirector, my external IP address 67.82.153.250Finding Slingbox on local network. My IP Info =  fe80::5d3c:7f28:4d0d:1303%6Finding Slingbox on local network. My IP Info =  ::1Finding Slingbox on local network. My IP Info =  127.0.0.1Using slingbox at  ('192.168.1.15', 5201)Streamer:  New York HD Slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingboxPING failure ('67.82.153.250', 8080) [WinError 10061] No connection could be made because the target machine actively refused itNo working external portmaps. Not registering slingboxesRetrying in 10 minutes09/15/2022, 22:22:53  Registering Slingboxes {'395808DB6C4591CE3DFA8E85F3309883': 8080}Getting external IP addressRedirector, my external IP address 67.82.153.250PING failure ('67.82.153.250', 8080) [WinError 10061] No connection could be made because the target machine actively refused itNo working external portmaps. Not registering slingboxesRetrying in 10 minutes

Couple things I see/confused about:it says:Checking for slingbox at ('http://192.168.1.15', 5201)Error connecting to  ('http://192.168.1.15', 5201)
and:PING failure ('67.82.153.250', 8080) [WinError 10061] No connection could be made because the target machine actively refused itNo working external portmaps. Not registering slingboxes

When i try to connect with VLC it never says it sees me connecting.
And after this is all over, PLEASE can i send you a digital gift card? Starbucks or something?
Ray

-----Original Message-----

**Reply by GerryDazoo** [reply]
OK, many things here.
1) Delete your finderid, you don't need it for now. That will get rid of your Ping errors. It's because you don't have the proper port forwarding and filrewall rules configured.
2) Your configured ip address and port number are wrong in the config.ini for your slingbox
You have  
ipaddress=http://192.168.1.15
port=5201/;
Should be ipaddress=192.168.1.5   No HTTP://
port=5201                                      No /;
That will remove the following error and all the logs about trying to find your slingbox on the network.
Checking for slingbox at ('http://192.168.1.15/', 5201)
Error connecting to  ('http://192.168.1.15/', 5201)

Please supply the exact url you are using for VLC. If it is correct then I would suspect your local firewall is blocking the connection. Turn it off temporarily to verify.

**Comment by RayMan412** [comment]
http://192.168.1.15:5201/slingbox
nothing yet but I think im close. No more errors on the streamer server. Going to see about the firewall


-----Original Message-----

**Reply by GerryDazoo** [reply]
That is the ip address and port number for the slingbox not the server!
try  http://192.168.1.21:8080/slingbox
make sure your firewall allows connections on port 8080

**Comment by RayMan412** [comment]
You are an absolute genius.
-Connected to that IP-Opened the firewall- Finally saw that it was trying to connect, but I had the wrong slingbox admin password.- Corrected the password.- Connected to the Ip....AND IT WORKED!!!
Now I'll have to configure remote access, but as long as I have this, I know I can do the rest.
Bought a new router that accepts No-Ip setup so I can have a static IP.
Thank you so so much!!
How can I send you a gift?
Ray

-----Original Message-----

**Comment by RayMan412** [comment]
Just getting an error trying to reach the remote
Starting remote control stream handler for  ('192.168.1.21', 60631) to port 8081Remote Control Connected127.0.0.1 - - [16/Sep/2022 12:11:04] "GET /remote HTTP/1.1" 404 -Exiting Remote Control Stream Handler for ('192.168.1.21', 60631)

-----Original Message-----

**Comment by GerryDazoo** [comment]
http://192.168.1.21:8080/Remote   ### Capital R

**Comment by RayMan412** [comment]
Which port (8080 I take it) should I be forwarding for ourside traffic and to which IP address?


-----Original Message-----

**Reply by GerryDazoo** [reply]
192.168.1.21
